packages/ckeditor5-fullscreen/src/handlers/abstracteditor.ts

```diff
@@ -177,4 +180,94 @@ export default class AbstractEditorHandler {
 
 		this.moveToFullscreen( documentOutlineUI.view.element!, 'document-outline' );
 	}
+
+	/**
+	 * Adds an event listener when the dialog opens to adjust its position in fullscreen mode,
+	 * utilizing the empty space on the right side of the editable element.
+	 */
+	public registerFullscreenDialogPositionAdjustements(): void {
+		if ( !this._editor.plugins.has( 'Dialog' ) ) {
+			return;
+		}
+
+		const dialog = this._editor.plugins.get( 'Dialog' ) as Dialog;
+
+		this.setNewDialogPosition();
+
+		dialog.on( 'change:isOpen', this.updateDialogPositionCallback, { priority: 'highest' } );
+	}
+
+	/**
+	 * Removes an event listener that adjusts the dialog's position in fullscreen mode.
+	 */
+	public unregisterFullscreenDialogPositionAdjustements(): void {
+		if ( !this._editor.plugins.has( 'Dialog' ) ) {
+			return;
+		}
+
+		const dialog = this._editor.plugins.get( 'Dialog' ) as Dialog;
+		const dialogView = dialog.view;
+
+		if ( dialogView && dialogView.position === null ) {
+			dialogView.position = DialogViewPosition.EDITOR_TOP_SIDE;
+		}
+
+		dialogView?.updatePosition();
```

Comment:
Unfortunately we're not supposed to use `?.` operator. I can't find the rationale right now so you have to trust me :D but I'll share it once I get it.